bots/flakes-refresh

```
@@ -0,0 +1,166 @@
+
+# How far back does our data go?  If a flake gets fixed but is still
+# flaky after this many days, the bots open another issue.
+
+WINDOW_DAYS = 90
+
+# This parses the output JSONL format discussed here, where various
+# values are grouped:
+#
+# https://github.com/cockpit-project/cockpituous/blob/master/learn/README.md
+
+# Here we're looking for a field in a record that only has one value
+def value(record, field):
+    values = record.get(field, [])
+    if len(values) == 1:
+        return values[0][0] or ""
+    return None
+
+# Here we're looking for the count of a specific field/value in the record
+def count(record, field, only):
+    values = record.get(field, [])
+    for value, count in values:
+        if value != only:
+            continue
+        return count
+    return 0
+
+def run(api, verbose=False, dry=False):
+    open_issues = api.issues(labels=[ "flake" ])
+    create_count = NUMBER_OPEN_ISSUES - len(open_issues)
+
+    if create_count <= 0:
+        return 0
+
+    if verbose:
+        sys.stderr.write("Going to create %s new flake issue(s)\n" % create_count)
+
+    host = os.environ.get("COCKPIT_LEARN_SERVICE_HOST")
+    if not host:
+        raise RuntimeError("no learn host environment variable: $COCKPIT_LEARN_SERVICE_HOST")
+    port = os.environ.get("COCKPIT_LEARN_SERVICE_PORT", "8080")
+    url = "http://{0}:{1}/active/statistics.jsonl".format(host, port)
+
+    statistics = [ ]
+
+    # Retrieve the URL
+    try:
+        req = urllib.request.Request(url)
+        with urllib.request.urlopen(req) as f:
+            for line in f.readlines():
+                try:
+                    record = json.loads(line.decode('utf-8'))
+                    statistics.append(record)
+                except ValueError as ex:
+                    sys.stderr.write("{0}: {1}\n".format(url, ex))
+    except (ConnectionResetError, urllib.error.URLError, socket.gaierror) as ex:
```

Fine; but might be easier to catch  their common subclass `OSError`.

---

I have removed this `try` block completely.  It should be fine to just let this crash.